File: packages/app-icon/android/src/main/java/io/capawesome/capacitorjs/plugins/appicon/classes/options/SetIconOptions.java
package io.capawesome.capacitorjs.plugins.appicon.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.appicon.classes.CustomExceptions;

public class SetIconOptions {

    @NonNull
    private final String icon;

    public SetIconOptions(@NonNull PluginCall call) throws Exception {
        this.icon = getIconFromCall(call);
    }

    @NonNull
    public String getIcon() {
        return icon;
    }

    @NonNull
    private static String getIconFromCall(@NonNull PluginCall call) throws Exception {
        String icon = call.getString("icon");
        if (icon == null) {
            throw CustomExceptions.ICON_MISSING;
        }
        return icon;
    }
}

File: packages/app-icon/ios/Plugin/Classes/Options/SetIconOptions.swift
import Foundation
import Capacitor

@objc public class SetIconOptions: NSObject {
    let icon: String

    init(_ call: CAPPluginCall) throws {
        self.icon = try SetIconOptions.getIconFromCall(call)
    }

    private static func getIconFromCall(_ call: CAPPluginCall) throws -> String {
        guard let icon = call.getString("icon") else {
            throw CustomError.iconMissing
        }
        return icon
    }
}
